pkg/manager/manager.go

```diff
@@ -142,18 +142,36 @@ type Options struct {
 	LeaderElection bool
 
 	// LeaderElectionResourceLock determines which resource lock to use for leader election,
-	// defaults to "configmapsleases". Change this value only if you know what you are doing.
+	// defaults to "leases". Change this value only if you know what you are doing.
 	// Otherwise, users of your controller might end up with multiple running instances that
 	// each acquired leadership through different resource locks during upgrades and thus
 	// act on the same resources concurrently.
-	// If you want to migrate to the "leases" resource lock, you might do so by migrating to the
-	// respective multilock first ("configmapsleases" or "endpointsleases"), which will acquire a
-	// leader lock on both resources. After all your users have migrated to the multilock, you can
-	// go ahead and migrate to "leases". Please also keep in mind, that users might skip versions
-	// of your controller.
 	//
-	// Note: before controller-runtime version v0.7, the resource lock was set to "configmaps".
-	// Please keep this in mind, when planning a proper migration path for your controller.
+	// If you are using `configmaps`/`endpoints` resource lock and want to migrate to "leases",
+	// you might do so by migrating to the respective multilock first ("configmapsleases" or "endpointsleases"),
+	// which will acquire a leader lock on both resources.
+	// After all your users have migrated to the multilock, you can go ahead and migrate to "leases".
+	// Please also keep in mind, that users might skip versions of your controller.
+	//
+	// Note: before controller-runtime version v0.7, it was set to "configmaps".
+	// And from v0.7 to v0.11, the default was "configmapsleases", which was
+	// used to migrate from configmaps to leases.
+	// Since the default was "configmapsleases" for over a year, spanning five minor releases,
+	// any actively maintained operators are very likely to have a released version that uses
+	// "configmapsleases". Therefore defaulting to "leases" should be safe since v0.12.
+	//
+	// So, what do you have to do when you are updating your controller-runtime dependency
+	// from a lower version to v0.12 or newer?
+	// - If your operator matches at least one of these conditions:
+	//   - the LeaderElectionResourceLock in your operator has already been explicitly set to "leases"
+	//   - the old controller-runtime version is between v0.7.0 and v0.11.x and the
```

Comment:
If the project was scaffold with <, 0.12 does this not mean that the default value is configmapsleases?
Then, would not also be requires to follow the same steps?

Reply:
Because "configmapsleases" can migrate to "leases" directly. It is designed as a middle state for projects migrating from "configmaps" to "leases".

Reply:
Hi @FillZpp,

I think I properly understood now.
Operators using c+r> 0.7 and <= 0.11 can safely only bump the controller-runtime to 0.12 because their lock is configured with configmapsleases. Therefore from configmapsleases to leases by default not breaking change or bad side effects can be faced. 

Am I right?

WDYT about we clarifies also the issues/bad side effects if someone does not follow the guidance?

Reply:
> Am I right?

Exactly.

> WDYT about we clarifies also the issues/bad side effects if someone does not follow the guidance?

Sure, I have moved the bad side effects under the steps.

Reply:
thank you for the clarifications